fmn/rules/utils.py

```diff
@@ -67,6 +72,32 @@ def get_fas(config):
     return _FAS
 
 
+def _paginate_pagure_data(url, params, key):
```

Comment:
Okay, I've documented this API, but I think we should change it.

For one thing, I don't like that errors are quietly swallowed. Things using this API may want to make decisions based on these errors like whether or not to cache the response. As a side note, the behavior of generators raising ``StopIteration`` [is about to change in Python 3.7](https://docs.python.org/3/library/exceptions.html#StopIteration). I only skimmed the associated [PEP](https://www.python.org/dev/peps/pep-0479/), but it seems like what is being done here is not correct (not 100% certain about that, though).

I also find the interface of passing a URL, the URL parameters, and a key I want from a JSON response odd. Why not just return the entire JSON object? Anyone who wants more than one key from the response is not going to enjoy using this.

I'd prefer to use pagination, but I'd also prefer us to not "roll our own" here when we could do it once in a Python binding library. @pypingou, is there an "official" Python bindings library?